airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -281,6 +304,49 @@ private void reportFailure(final ConnectionUpdaterInput connectionUpdaterInput,
     }
   }
 
+  private StandardSyncOutput checkConnections(GenerateInputActivity.GeneratedJobInput jobInputs) {
+    final int attemptCreationVersion =
+        Workflow.getVersion(CHECK_BEFORE_SYNC_TAG, Workflow.DEFAULT_VERSION, CHECK_BEFORE_SYNC_CURRENT_VERSION);
+
+    if (attemptCreationVersion < CHECK_BEFORE_SYNC_CURRENT_VERSION) {
+      return null;
+    }
+
+    final StandardCheckConnectionInput sourceConfiguration = new StandardCheckConnectionInput()
+        .withConnectionConfiguration(jobInputs.getSyncInput().getSourceConfiguration());
+    final CheckConnectionInput checkSourceInput =
+        new CheckConnectionInput(jobInputs.getJobRunConfig(), jobInputs.getSourceLauncherConfig(), sourceConfiguration);
+    final String launchSourceDockerImage = jobInputs.getSourceLauncherConfig().getDockerImage();
```

Comment:
This was fascinating - the docker image that will be run changes if it's a RESET to an image that doesn't exist (`airbyte_empty`).  This will fail the CHECK for sure, and should be skipped.  Of note, `getDockerImage()` from `SourceLauncherConfig()` returns the "running" docker image, and calling the same method on `SyncInput()` contains the original docker image from the connector.